ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/SavePreMergeHeadersStep.java

```diff
@@ -51,11 +59,16 @@ public Stream<BlockHeader> apply(final BlockHeader blockHeader) {
     }
     storeBlockHeader(blockHeader);
     logProgress(blockHeader);
+    if (blockHeader.getNumber() == lastPoWBlockNumber) {
+      blockchain
```

Comment:
nit: move to `storeBlockHeader`

Reply:
done in [`cdd3319` (#8634)](https://github.com/hyperledger/besu/pull/8634/commits/cdd3319530e90124e79b0e555c0be9342bb33872)